File: 169-MajorityElement/169-MajorityElement.java
// Last updated: 7/14/2026, 2:04:20 PM
class Solution {
    public int majorityElement(int[] nums) {
        int candidate = 0;
        int count = 0;
        
        
        for (int num : nums) {
            if (count == 0) {
                candidate = num;  
            }
            count += (num == candidate) ? 1 : -1; 
        }
        
        return candidate;
    }
}

File: 169-MajorityElement/169-MajorityElement.py
# Last updated: 7/14/2026, 2:04:15 PM
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        n=floor(len(nums)/2)
        count=0
        candidate=None
        for i in nums:
            if count==0:
                candidate=i
            if i==candidate:
                count+=1
            else:
                count-=1
        return candidate